src/main/java/cn/nukkit/block/BlockIce.java

```diff
@@ -40,15 +42,22 @@ public int getToolType() {
     }
 
     @Override
-    public boolean onBreak(Item item) {
-        this.getLevel().setBlock(this, Block.get(WATER), true);
-        return true;
+    public boolean onBreak(Item item, Player player) {
+        if (player.getGamemode() == Player.CREATIVE) {
+            return this.getLevel().setBlock(this, Block.get(AIR), true);
+        }
+
+        if (down().isSolid()) {
+            return this.getLevel().setBlock(this, Block.get(WATER), true);
+        } else {
+            return this.getLevel().setBlock(this, Block.get(AIR), true);
+        }
```

Comment:
So using the new method, we can now check if the player is in creative and destroy the block directly in this case.

An other change here is that the ice will only become water if the block under it is solid, the same behaviour as vanilla Minecraft.